app/src/main/java/com/radixdlt/android/ui/fragment/MoreOptionsFragment.kt

```diff
@@ -207,26 +227,57 @@ class MoreOptionsFragment : Fragment() {
     override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
         if (resultCode != Activity.RESULT_OK) return
 
-        if (requestCode == REQUEST_CODE_SET_TIME_OUT) {
-            autoLockTimeOutTimeTextView.text = displayAutoLockTime()
-        } else if (requestCode == REQUEST_CODE_DELETE_WALLET) {
-            Vault.resetKey()
-            QueryPreferences.setPrefAddress(activity!!, "")
-            QueryPreferences.setPrefPasswordEnabled(activity!!, true)
-            QueryPreferences.setPrefAutoLockTimeOut(activity!!, 2000)
-            Identity.clear()
-            val myKeyFile = File(activity!!.filesDir, "keystore.key")
-            myKeyFile.delete()
-
-            Completable.fromAction(::deleteTables)
-                .subscribeOn(Schedulers.single())
-                .subscribe()
-
-            activity!!.startActivity<NewWalletActivity>()
-            activity!!.finish()
+        when (requestCode) {
+            REQUEST_CODE_SET_TIME_OUT -> autoLockTimeOutTimeTextView.text = displayAutoLockTime()
+            REQUEST_CODE_DELETE_WALLET -> {
+                resetData()
+                deleteKeystoreFile()
+
+                activity!!.startActivity<NewWalletActivity>()
+                activity!!.finish()
+            }
+            REQUEST_CODE_CHOOSE_NETWORK -> {
+                universe = data!!.getIntExtra(ChooseNetworkDialog.EXTRA_UNIVERSE, 0)
+
+                val warningDialog = WarningDialog.newInstance()
+                warningDialog.setTargetFragment(
+                    this@MoreOptionsFragment, REQUEST_CODE_WARNING
+                )
+                warningDialog.show(fragmentManager, "WARNING_DIALOG")
+            }
+            REQUEST_CODE_WARNING -> {
+                if (universe == 0) {
+                    QueryPreferences.setPrefNetwork(activity!!, ALPHANET)
+                } else {
+                    QueryPreferences.setPrefNetwork(activity!!, ALPHANET2)
+                }
+
+                resetData()
+                ProcessPhoenix.triggerRebirth(activity)
+            }
         }
     }
 
+    private fun resetData() {
+        Completable.fromAction { Vault.resetKey() }
+            .subscribeOn(Schedulers.computation())
+            .subscribe()
+
+        QueryPreferences.setPrefAddress(activity!!, "")
+        QueryPreferences.setPrefPasswordEnabled(activity!!, true)
+        QueryPreferences.setPrefAutoLockTimeOut(activity!!, 2000)
+        Identity.clear()
+
+        Completable.fromAction(::deleteTables)
+            .subscribeOn(Schedulers.io())
+            .subscribe()
+    }
+
+    private fun deleteKeystoreFile() {
+        val myKeyFile = File(activity!!.filesDir, "keystore.key")
+        myKeyFile.delete()
+    }
```

Comment:
Is there any particular reason this is separated from `resetData()`? Would we ever want to do one without the other? If not I'd suggest we merge these two methods.

Reply:
When changing the universe or connecting to a node I need to reset most data but keystore is not necessary to delete. I will refactor this anyways in a util/helper class and create a specific method when resetting/cleaning everything is necessary!